lib/ansible/modules/cloud/vmware/vmware_guest_attribute.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright, (c) 2018, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: vmware_guest_attribute
+short_description: Manage custom attributes from VMWare for virtual machine
+description:
+    - This module can be used to add, remove, list custom attributes for the given virtual machine.
+version_added: 2.6
+author:
+    - Jimmy Conner
+    - Abhijeet Kasurde (@akasurde) <akasurde@redhat.com>
+notes:
+    - Tested on vSphere 6.5
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+   name:
+     description:
+     - Name of the virtual machine to work with.
+     required: True
+   state:
+     description:
+     - The action to take.
```

Comment:
Think this requires a bit more meat... Something like `Determines the action to be taken, when present, the custom value will be added or updated. When absent, it will be removed.`

Should list move to its own facts module?